File: Leetcode 3701-3800/3731/3731.java
// Leetcode 3731: Find Missing Elements
// https://leetcode.com/problems/find-missing-elements/
// Solved on 4th of August, 2026
import java.util.ArrayList;
import java.util.List;

class Solution {
    /**
     * Finds the missing elements in an array of integers.
     * 
     * @param nums The array of integers.
     * @return A list of missing integers.
     */
    public List<Integer> findMissingElements(int[] nums) {
        int minVal = nums[0];
        int maxVal = nums[0];
        for (int num : nums) {
            if (num < minVal) {
                minVal = num;
            }
            if (num > maxVal) {
                maxVal = num;
            }
        }
        boolean[] present = new boolean[maxVal + 1];
        for (int num : nums) {
            present[num] = true;
        }
        List<Integer> result = new ArrayList<>();
        for (int i = minVal; i <= maxVal; i++) {
            if (!present[i]) {
                result.add(i);
            }
        }
        return result;
    }
}

File: Leetcode 3701-3800/3731/3731.py
# Leetcode 3731: Find Missing Elements
# https://leetcode.com/problems/find-missing-elements/
# Solved on 4th of August, 2026
class Solution:
    def findMissingElements(self, nums: list[int]) -> list[int]:
        """
        Finds the missing elements in an array of integers.
        
        @param nums: The array of integers.
        @return: A list of missing integers.
        """
        numSet = set(nums)

        minVal = min(nums)
        maxVal = max(nums)

        missingNums = []
        for currentNum in range(minVal, maxVal + 1):
            if currentNum not in numSet:
                missingNums.append(currentNum)

        return missingNums